### PR TITLE
Fix projectile interaction with obstacles; remove ACID_DROP from documentation

### LINKED ISSUE #17709: Corrosives can shoot acid through impassable stuff
They can shoot acid through both impassable vehicle parts and impassable furniture/terrain. Is this intended?

### PR DESCRIPTION
**Pull request by sethsimon**
Fixes #17709.

Non-bullet projectiles will only penetrate an impassable obstacle if it destroys the obstacle and still has momentum afterwards (e.g. throwing axe vs. window).  Bullet projectiles (those that deal DT_CUT and have a high speed) will penetrate an impassable obstacle whether it destroys the obstacle or not, as long as it has momentum after hitting it.

Also, it's a little odd that the acid that corrosives spit disappears if it hits the ground (and not a creature). I noticed that there is a documented ammo effect in `doc/JSON_FLAGS.md` called `ACID_DROP` that "creates a tiny field of weak acid." Strangely, it wasn't implemented, so I implemented it and added it to the projectile created by `mattack::acid_accurate`.

This has the effect of buffing corrosives a bit. In testing, it appears that if you leave the field as soon as it appears under your feet, the field will cause about 10 damage if you have no clothes on your legs or feet, or 5 damage when wearing socks/boots and briefs/cargo pants.

### CODE REVIEW

**Comment by pisskop** [comment]
im not sure cortosives spit enough acid to make a puddle.

in fact, im sure it was removed because its annoying unavoidable damage and encourages minmaxing.

corrosive acid doesnt puddle when it hits a player, unless thats been readded very recently

**Comment by sethsimon** [comment]
You're right, it was removed in #14775.

File: src/ranged.cpp [hunk 1]
```diff
@@ -141,6 +141,13 @@ dealt_projectile_attack Creature::projectile_attack( const projectile &proj_arg,
     const bool no_item_damage = proj_effects.count( "NO_ITEM_DAMAGE" ) > 0;
     const bool do_draw_line = proj_effects.count( "DRAW_AS_LINE" ) > 0;
     const bool null_source = proj_effects.count( "NULL_SOURCE" ) > 0;
+    // Determines whether it can penetrate obstacles
+    const bool is_bullet = proj_arg.speed >= 200 && std::any_of( proj_arg.impact.damage_units.begin(),
+                           proj_arg.impact.damage_units.end(),
+                           []( const damage_unit &dam )
+    {
+        return dam.type == DT_CUT;
```
Comment (Coolthulhu) [inline src/ranged.cpp]: This will make thrown rocks stop at windows.

File: src/projectile.cpp [hunk 2]
```diff
@@ -136,6 +136,10 @@ void apply_ammo_effects( const tripoint &p, const std::set<std::string> &effects
         }
     }
 
+    if( effects.count( "ACID_DROP" ) > 0 ) {
+        g->m.add_field( p, fd_acid, 1, 0 );
+    }
```
Comment (Coolthulhu) [inline src/projectile.cpp]: Now it's not used anywhere

File: src/ranged.cpp [hunk 3]
```diff
@@ -141,6 +141,13 @@ dealt_projectile_attack Creature::projectile_attack( const projectile &proj_arg,
...
+    } );
```
Comment (Coolthulhu) [inline src/ranged.cpp]: This prevents thrown rocks from going through windows

File: src/projectile.cpp (hunk 2 above)
Comment (sethsimon) [inline src/projectile.cpp]: What if someone wants to use it in their mod?

Comment (mugling) [inline src/projectile.cpp]: Needs a consumer in `core` or at least within `master` or it's just going to bitrot.

Comment (mugling) [inline src/projectile.cpp]: The long term plan is to support `emit` for ammo so that mod authors can define fields in JSON

Comment (codemime) [inline src/projectile.cpp]: Use what? There's no such thing as `ACID_DROP` anywhere in the project (neither in this PR).

File: src/ranged.cpp (hunk 1 above)
Comment (sethsimon) [inline src/ranged.cpp]: IRL, a rock that's badly aimed, weakly thrown, or thrown from far away might not go through a window. With 2 throwing, 5 strength, 5 dexterity, and from 1 tile away, 4 out of 14 throws broke a window. With 2 throwing, 8 strength, and 8 dexterity, you can reliably break it from 3 tiles away. With 0 throwing, 8 strength, and 8 dexterity, you can reliably break a window from 2 tiles away. With 1 throwing, 5 strength, and 5 dexterity, it's very difficult (but possible) to break a window from 1 tile away.

File: src/projectile.cpp (hunk 2 above)
Comment (sethsimon) [inline src/projectile.cpp]: > Use what? There's no such thing as ACID_DROP anywhere in the project (neither in this PR).

It was documented in `JSON_FLAGS.md`. But due to its lack of consumers, I'll remove both the documentation and implemenation.

**Pull request merged**